src/fairchem/core/common/distutils.py

```diff
@@ -210,3 +218,31 @@ def gather_objects(data: T, group: dist.ProcessGroup = dist.group.WORLD) -> list
     output = [None for _ in range(get_world_size())] if is_master() else None
     dist.gather_object(data, output, group=group, dst=0)
     return output
+
+
+def assign_device_for_local_rank(cpu: bool, local_rank: int):
+    if cpu:
+        os.environ[CURRENT_DEVICE_STR] = "cpu"
+    else:
+        # assert the cuda device to be the local rank
+        os.environ[CURRENT_DEVICE_STR] = "cuda"
+        os.environ["CUDA_VISIBLE_DEVICES"] = str(local_rank)
```

Comment:
Does this line do anything? is this intended to affect child processes?

Reply:
ya this is the safer way to assign a cuda device to a process, instead using torch.cuda.set_device, we make each process only see a single GPU with id cuda:0, this way no process need to know about their device id